gobblin-utility/src/main/java/org/apache/gobblin/util/retry/RetryerFactory.java

```diff
@@ -87,15 +97,18 @@ public static <T> Retryer<T> newInstance(Config config, Optional<RetryListener>
     RetryType type = RetryType.valueOf(config.getString(RETRY_TYPE).toUpperCase());
 
     RetryerBuilder<T> builder;
+
+    Predicate<Throwable> retryPredicate = getRetryPredicateFromConfigOrDefault(config);
+
     switch (type) {
       case EXPONENTIAL:
-        builder = newExponentialRetryerBuilder(config);
+        builder = newExponentialRetryerBuilder(config, retryPredicate);
         break;
       case FIXED:
-        builder = newFixedRetryerBuilder(config);
+        builder = newFixedRetryerBuilder(config, retryPredicate);
```

Comment:
lets not add something to the param that can be constructer using the existing param.
lets move `Predicate<Throwable> retryPredicate = getRetryPredicateFromConfigOrDefault(config);` inside `newFixedRetryerBuilder` and other builders

Reply:
addressed